llvm/utils/repo/Makefile

```diff
@@ -65,24 +69,24 @@ libcompiler-rt.a:
 		-D CMAKE_CXX_COMPILER=$(LLVM_BUILD)/bin/clang++               \
 		-D CMAKE_C_COMPILER=$(LLVM_BUILD)/bin/clang                   \
 		-D utils_dir=$(LLVM_REPO)/llvm/utils/repo                     \
-		-D CMAKE_INSTALL_PREFIX=$(LLVM)                               \
+		-D CMAKE_INSTALL_PREFIX=$(CLANG_RT)                           \
 		-D musl_install=$(MUSL)                                       \
 		../compiler-rt &&                                             \
 	make -j 8 &&                                                          \
 	make install
 
-$(LLVM)/lib/linux/clang_rt.crtbegin-x86_64.o: libcompiler-rt.a
-$(LLVM)/lib/linux/clang_rt.crtend-x86_64.o: libcompiler-rt.a
+$(CLANG_RT)/lib/linux/clang_rt.crtbegin-x86_64.o: libcompiler-rt.a
+$(CLANG_RT)/lib/linux/clang_rt.crtend-x86_64.o: libcompiler-rt.a
 
-$(LLVM)/lib/linux/clang_rt.crtbegin-x86_64.o.elf: $(LLVM)/lib/linux/clang_rt.crtbegin-x86_64.o
-	repo2obj --repo=$(LLVM_REPO)/build_libcompiler-rt/libcompiler-rt.db $< -o $@
+$(CLANG_RT)/lib/linux/clang_rt.crtbegin-x86_64.t: $(CLANG_RT)/lib/linux/clang_rt.crtbegin-x86_64.o
+	mv  $< $@
+	repo2obj --repo=$(LLVM_REPO)/build_libcompiler-rt/libcompiler-rt.db $@ -o $<

```

Comment:
I'm baffled by this rule. 

The rule will fire when the .o file is newer than the .t. It seems to first rename the .o as .t then run repo2obj to overwrite the newer file with the contents of the older.

Reply:
Yes,  you are right. The rule only is invoked when targeting on the ELF using repo2obj.

The Makefile could be used when targeting on ELF (using repo2obj) or on Repo. By default, it is targeted on the ELF. Since the clang driver linked the _clang_rt.crtbegin-x86_64.o_ file, therefore we expect:
1) By default, when targeting on ELF using repo2obj, the clang_rt.crtbegin-x86_64.o is **ELF** format and clang_rt.crtbegin-x86_64.t is the ticket format. In this case, the above rule will be used.
2) When targeting on Repo by passing FULL_REPO=1, the clang_rt.crtbegin-x86_64.o is **Repo** format. In this case, the above rule will not be invoked.

Reply:
Understood, but I don’t understand the change in the the way that the rule is written. If you want to produce a target ELF file (presumably the .o extension) from a ticket prerequisite (the .t), surely the rule must be written the other way round (i.e. `x.o: x.t`) as it is now.

(Comments apply to the crtend-x86_64 rule as well, of course.)

Reply:
There are two existing rules:
```
$(LLVM)/lib/linux/clang_rt.crtbegin-x86_64.o: libcompiler-rt.a
$(LLVM)/lib/linux/clang_rt.crtend-x86_64.o: libcompiler-rt.a
```
I can’t add other two rules as you suggested (x.o: x.t) since the rule x.o already exists.

Any suggestion? 
Thanks,

Reply:
Why do these two files depend on libcompiler-rt.a? There seem to be no associated commands.

Reply:
>
> Why do these two files depend on libcompiler-rt.a?
>

These two files include in the compiler-rt project. When you build compiler-rt project on Repo, these two files *.o are compiled on the ticket format, then I want to rename them to *.t and convert them to the *.o files using repo2obj. In summary, the first two *.o are Repo ticket format and the last two *.o are ELF format. 

Updated the patch in commit https://github.com/SNSystems/llvm-project-prepo/commit/19a47bdab5b8aa4dd46e86ac70fee194aa4fdd00.